subprojects/base-services/src/main/java/org/gradle/internal/resources/DefaultResourceLockCoordinationService.java

```diff
@@ -125,12 +125,13 @@ boolean hasUnlockedResources() {
 
         @Override
         public void releaseLocks() {
-            if (lockedResources != null) {
+            if (lockedResources != null && !lockedResources.isEmpty()) {
```

Comment:
This is not really necessary - the for loop will just short-circuit anyway.

Reply:
I removed it again.